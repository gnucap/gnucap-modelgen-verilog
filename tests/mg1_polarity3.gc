
attach ./modelgen_0.so


verilog

`modelgen
module polarity3_0(p, n);
	electrical p, n;
	inout p, n;
	real VV;
	analog begin : main
		real i;
		VV = 2.0 * V(n, p);
		i = V(p,n);
		I(p,n) <+ i;
	end
endmodule

list

!make polarity3_0.so > /dev/null
attach ./polarity3_0.so

polarity3_0 dut(1,0);
vsource v1(1, 0);

list

print dc v(v1) i(v1) r(dut.*) iter(0)

dc v1 0 2 1
