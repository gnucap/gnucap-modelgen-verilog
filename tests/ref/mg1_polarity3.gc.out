polarity3_0 #() dut (.p(1),.n(0));
vsource #() v1 (.p(1),.n(0));
#           v(v1)      i(v1)      r(dut._b_n_p) iter(0)   
 0.         0.         0.         1.         3.        
 1.         0.99999   -0.99999    1.         3.        
 2.         2.        -2.         1.         3.        
